src/System.Net.Security/tests/FunctionalTests/SslStreamStreamToStreamTest.cs

```diff
@@ -85,6 +85,63 @@ await TestConfiguration.WhenAllOrAnyFailedWithTimeout(client.AuthenticateAsClien
             }
         }
 
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, ".NET Framework does not recover well from underlying stream failures")]
+        [Fact]
+        public async Task Read_CorrectlyUnlocksAfterFailure()
+        {
+            var network = new VirtualNetwork();
+            var clientStream = new ThrowingDelegatingStream(new VirtualNetworkStream(network, isServer: false));
+            using (var clientSslStream = new SslStream(clientStream, false, AllowAnyServerCertificate))
+            using (var serverSslStream = new SslStream(new VirtualNetworkStream(network, isServer: true)))
+            {
+                await DoHandshake(clientSslStream, serverSslStream);
+
+                // Throw an exception from the wrapped stream's read operation
+                clientStream.ExceptionToThrow = new FormatException();
+                IOException thrown = await Assert.ThrowsAsync<IOException>(() => ReadAsync(clientSslStream, new byte[1], 0, 1));
+                Assert.Same(clientStream.ExceptionToThrow, thrown.InnerException);
+                clientStream.ExceptionToThrow = null;
+
+                // Validate that the SslStream continues to be usable
+                for (byte b = 42; b < 52; b++)
```

Comment:
What is the significance of the magic number "42" and "52"??

Reply:
Nothing.  I just wanted to run it a few times and start with a non-zero value.  I often pick "42" for that out of habit.

Reply:
To avoid questions from other code reviewers and future developers, it would be helpful to add a comment such as:

```c#
for (byte b = 42; b < 52; b++) // byte values chosen at random
```

Reply:
Done. Thanks.